### DISCUSSION TITLE
DNS-O-Matic username does not allow for emails

### DISCUSSION THREAD

**Discussion by josemh** [qdm12/ddns-updater#398] [Q&A] [answered]
It seems that OpenDNS login with a username is not working. I have to use an email instead of a username to login. This breaks the ddns-updater validation rule for dnsomatic username. It appears that ddns-omatic does not like the '@' symbol in the username.

The regex to validate the dnsomatic username in the regex.go file is:
`dnsOMaticUsername        = regexp.MustCompile('^[a-zA-Z0-9._-]{3,25}$')`

**Comment by qdm12** [comment, marked as answer]
Indeed! Fixed in 59bffe99ba494011963fc0bf5f9bb3ae8e01f3ff thanks for taking the time to report it.

**Reply by DieSchoeneWolke** [reply]
Hey Quentin, I guess the regexp is missing a + for filter aliases e.g. from Gmail.
https://gmail.googleblog.com/2008/03/2-hidden-ways-to-get-more-from-your.html

`ERROR username is not valid: username "...+...@gmail.com" does not match regex "^[a-zA-Z0-9@._-]{3,25}$"`

**Reply by qdm12** [reply]
Fixed in 496781ac34fa04cfc07c05eeeeca1261da5ee476 thanks 👍

**Reply by DieSchoeneWolke** [reply]
Ok, I built a new image today, but unfortunately I still get the error message, but with a `+` included.

`ERROR username is not valid: username "...+...@gmail.com" does not match regex "^[a-zA-Z0-9+@._-]{3.25}$"`

The dnsomatic website changed some time ago. To log in you need to visit https://login.opendns.com.
As mentioned in their blog here: https://support.opendns.com/hc/en-us/articles/360037648951

I'm not sure by looking at the dnsomatic function in the code if it takes this into account.
https://github.com/qdm12/ddns-updater/blob/ba2e77798a569ddcf5ec1edb5608f2f7c491bd67/internal/provider/providers/dnsomatic/provider.go#L97

**Reply by qdm12** [reply]
sorry for the mega delay, I removed the user check completely so it should now work.
